src/main/drivers/dshot.c

```diff
@@ -139,6 +141,25 @@ uint16_t getDshotTelemetry(uint8_t index)
     return dshotTelemetryState.motorState[index].telemetryValue;
 }
 
+uint16_t erpmToRpm(int erpm)
+{
+    //  rpm = (erpm * 100) / (motorConfig()->motorPoleCount / 2)
+    return (erpm * 200) / motorConfig()->motorPoleCount;
+}
+
+// Caller shall ensure that dshot telemetry is active before calling this function (i.e. erpm data has already received)
+uint16_t getDshotAverageRpm(void)
+{
+    const int motor_count = getMotorCount();
+    uint32_t rpm = 0;
+
+    for (int i = 0; i < motor_count; i++) {
+        rpm += dshotTelemetryState.motorState[i].telemetryValue;
+    }
+
+    return erpmToRpm(rpm / motor_count);
+}
```

Comment:
```suggestion
uint16_t getDshotAverageRpm(void)
{
    return averageRpm;
}
```

Calculations in getter functions are problematic. Depending on how often we want to read the average RPM and how many motors are active, this can become quite inefficient.

We should rather let this code run in a loop and update a static variable once every loop (for example `static uint16_t averageRpm`), so that we can then call `getDshotAverageRpm()` as often as we want within a loop to return `averageRpm`.

Right now we might already call this whole calculation twice per loop and waste CPU cycles.

Reply:
Reworked